Main/Workshop3/Game.cpp

```diff
@@ -0,0 +1,242 @@
+    } else {
+        std::cout << "You are trying flee and roll your dice.\n";
+        std::uniform_int_distribution<uint_fast8_t> fleeDist(1, 6);
+        uint_fast8_t const                          fleeRoll = fleeDist(rng);
+        std::cout << "You've rolled a " << +fleeRoll << ".\n";
+        if (fleeRoll > 4) {
+            std::cout << "You've successfully fled!\n";
+            std::uniform_int_distribution<uint_fast8_t> runawayDist(1, 3);
+            uint_fast8_t const                          runawayRoll = runawayDist(rng);
+            std::unique_ptr<RunawayPolicy>              policy;
+            if (runawayRoll == 1) {
+                policy = std::make_unique<Runaway_ItemEquippedRemoval>();
+            } else if (runawayRoll == 2) {
+                policy = std::make_unique<Runaway_ModifierFromHandRemoval>();
+            } else {
+                policy = std::make_unique<Runaway_BiggestBonusCardRemoval>();
+            }
+            std::cout << policy->getFullInfo() << std::endl;
+            policy->apply(*this);
+        } else {
+            std::cout << "You died.\n";
+        }
+    }
+    return false;
+}
+
+
+bool Game::equipItem(uint_fast8_t index) {
+    if (index >= itemHand.size()) {
+        std::cout << "Invalid item index\n";
+        return false;
+    }
+    equippedItem = std::move(itemHand[index]);
+    itemHand.erase(itemHand.begin() + index);
+    std::cout << "Equipped item: " << equippedItem->getName() << std::endl;
+    return true;
+}
+
+void Game::removeEquippedItem() {
+    if (!equippedItem) {
+        std::cout << "Tried to remove an equipped item, but there was none.\n";
+        return;
+    }
+    std::cout << "Removed equipped item: " << equippedItem->getName() << std::endl;
+    if (equippedItem) equippedItem.reset();
+}
+
+void Game::removeRandomModifierFromHand() {
+    if (modifierHand.empty()) {
+        std::cout << "Tried to remove a modifier from your hand, but there was none.\n";
+        return;
+    }
+    static thread_local std::mt19937      rng(std::random_device{}());
+    std::uniform_int_distribution<size_t> dist(0, modifierHand.size() - 1);
+    size_t const                          index = dist(rng);
+    std::cout << "Removed modifier from hand: " << modifierHand[index]->getName() << std::endl;
+    modifierHand.erase(modifierHand.begin() + index);
+}
+
+void Game::removeBiggestBonusCard() {
+    if (itemHand.empty()) {
+        std::cout << "Tried to remove the biggest bonus card, but there was none.\n";
+        return;
+    }
+    auto it = std::max_element(itemHand.begin(), itemHand.end(),
+                               [](const std::unique_ptr<Item>& a, const std::unique_ptr<Item>& b) {
+                                   return a->getBonus() < b->getBonus();
+                               });
+    std::cout << "Removed the biggest bonus card: " << ( *it )->getName() << " (Bonus: " << +( *it )->getBonus()
+              << ")\n";
+    itemHand.erase(it);
+}
```

Comment:
Hmmm, but why would you delegate this stuff to Game itself? It's can be cleary part of the player class.